Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift

```diff
@@ -167,6 +167,27 @@ public extension Sequence {
         return try filter { set.insert(try transform($0)).inserted }
     }
 
+    /// SwifterSwift: Separates all items into 2 lists based on a given predicate.
+    /// The first list contains all items for which the specified condition evaluates to true.
+    /// The second list contains those that don't.
+    ///
+    ///     let (even, odd) = [0, 1, 2, 3, 4, 5].divided { $0 % 2 == 0 }
+    ///     let (minors, adults) = people.divided { $0.age < 18 }
+    ///
+    /// - Parameter condition: condition to evaluate each element against.
+    /// - Returns: A tuple of matched and non-matched items
+    func divided(by condition: (Element) throws -> Bool) rethrows -> (matching: [Element], nonMatching: [Element]) {
+        //Inspired by: http://ruby-doc.org/core-2.5.0/Enumerable.html#method-i-partition
+        var matching = ContiguousArray<Element>()
+        var nonMatching = ContiguousArray<Element>()
+        
+        var iterator = self.makeIterator()
+        while let element = iterator.next() {
+            try condition(element) ? matching.append(element) : nonMatching.append(element)
+        }
+        return (Array(matching), Array(nonMatching))
```

Comment:
Instead of using a contiguous array and then converting here, we could just use an array :))

Reply:
I just thought of using ContiguousArray to be more efficient and have more predictable performance than Array. I've just mimicked how Swift implements it. https://github.com/apple/swift/blob/master/stdlib/public/core/Sequence.swift#L646 Haha!

Reply:
Shall we use `[Element]` instead? :D

Reply:
Nice, I just thought that wouldn't be worth to pay the price for the creation of a new array. But makes sense to use Contiguous array for performance :)) So let's keep it that way

Reply:
Coolio 👍